sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -528,4 +468,53 @@ namespace Azure { namespace Core { namespace Http {
     std::vector<uint8_t> const& GetBody() const { return this->m_body; }
   };
 
+  namespace _detail {
+    struct RawResponse
```

Comment:
why?  its the same thing

Reply:
We decided that `struct` would only be used for holders of "properties", and any object that has methods and ctors would be a `class`. Let me see if that ended up in the C++ guidelines.

Regarding the name, I don't think we should use the identical name as the public type (`RawResponse`), as it adds confusion at the call-site.

Reply:
https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-design-naming-classes-no-struct-keyword

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/2002

Reply:
It has public member functions. Structs are for types that only have **public data members**.

That means options, model types, and things like `HttpRange`/`MatchConditions` should be struct, and everything else becomes class.

We have a couple of private/implementation detail types within the repo that currently don't follow this, but we should fix those.

Reply:
There's no benefit at all from such a guideline,  we should not have something like that.
The only difference from struct and class is the default access for it. 
Specially if we are disabling having methods directly under a namespace like ::_detail::Foo() and we expect to always use a Container for the methods like ::_detail::Container::Foo() , then all methods will be public, and class is just weird..

Reply:
> class is just weird..

Why?

> There's no benefit at all from such a guideline, we should not have something like that.

Well, if you have concerns about the guidelines, let's have that guideline discussion separately. @RickWinter should we change our guideline on when we use `struct` vs `class`?

Up to you what you want to do here, it's your PR. Please rename the type though.

Reply:
> won't give you nothing but an extra line of code.

I really wish we stop revisiting closed discussions in PRs like this (especially relitigating using the exact same reasons that were already settled, rather than providing new information/insight to motivate the change). I was assuming @RickWinter had updated the guideline after we discussed and agreed on when we would use `class` and `struct`.

My suggestion of the code change is based on what I understood to be the guideline, if you disagree, ignore my feedback. My intent is not to keep defending the guideline.